### PR TITLE
[opentelemetry-kube-stack] Fix values.schema.json for accepting targetAllocator.image type

### LINKED ISSUE #1250: [opentelemetry-kube-stack] Inconsistent TargetAllocator Schema
I found inconsistencies between the stack `values.schema.json` and the CRD. When attempting to add the following lines into charts/opentelemetry-kube-stack/examples/cloud-demo:
```
  targetAllocator:
    image: ghcr.io/open-telemetry/opentelemetry-operator/target-allocator:main
    enabled: true
    prometheusCR:
      enabled: true
      serviceMonitorSelector:
        matchLabels:
          app: my-app
```

the `make generate-examples` command raised the following errors:
```
Generating example: cloud-demo
Error: values don't meet the specifications of the schema(s) in the following chart(s):
opentelemetry-kube-stack:
- collectors.defaultCRConfig.targetAllocator.image: Invalid type. Expected: object, given: string
- collectors.defaultCRConfig.targetAllocator.prometheusCR.serviceMonitorSelector.matchLabels: Invalid type. Expected: string, given: object
```

Additionally, the line `"$id": "https://github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1/open-telemetry-collector-spec"`, in `values.schema.json` seems irrelevant. The stack generates apiVersion: opentelemetry.io/v1beta1 and the file in the provided URL does not exist.

### PR DESCRIPTION
**Pull request by sergeybataev**
Fix #1250

### CODE REVIEW

**Comment by jaronoff97** [comment]
@sergeybataev thanks for the PR! you'll need to run `make generate-examples CHARTS=opentelemetry-kube-stack` to ensure this works as expected.

**Comment by sergeybataev** [comment]
@jaronoff97 done

**Comment by JaredTan95** [comment]
I think the correct fix should be to follow the scheme registry/tepository:tag design and update related tpl, not change it to string.

File: charts/opentelemetry-kube-stack/values.schema.json [hunk 1]
```diff
@@ -1606,27 +1605,7 @@
           "type": "string"
         },
         "image": {
-          "description": "Image use in both standalone and agent configs",
```
Comment (JaredTan95) [inline charts/opentelemetry-kube-stack/values.schema.json:1609]: I think the correct fix should be to follow the scheme registry/tepository:tag design and update related tpl, not change it to string.

**Review by jaronoff97 — APPROVED** [review]

File: charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+      enabled: true
+      podMonitorSelector:
+        matchLabels:
+          app: my-app
+      scrapeInterval: "30s"
+      serviceMonitorSelector:
+        matchExpressions:
+          - key: kubernetes.io/app-name
+            operator: In
+            values: ["my-app"]
```
Comment (jaronoff97) [inline charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml:17]: @sergeybataev would you mind moving this to a new example? I'm trying to have this example be something a user would install to get started with otel and a new example would be great for the prometheus specific use case. I'm going to be working on this for #1080

**Comment by jaronoff97** [comment]
I am working through the CI failures as well. I'm not sure what's going wrong here.

**Comment by jaronoff97** [comment]
Ah i see, you need to run `make generate-examples` again after the version bump

**Comment by sergeybataev** [comment]
I created new example just for targetAllocator.

File: charts/opentelemetry-kube-stack/examples/targetallocator-prom/values.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
+clusterName: demo
+opentelemetry-operator:
+  enabled: true
+defaultCRConfig:
+  targetAllocator:
+    image: ghcr.io/open-telemetry/opentelemetry-operator/target-allocator:main
```
Comment (jaronoff97) [inline charts/opentelemetry-kube-stack/examples/targetallocator-prom/values.yaml:6]: note: I think we should set `enabled: true` here

**Review by jaronoff97 — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Comment by jaronoff97** [comment]
@sergeybataev Thank you for contribution!!

**Pull request merged**